sql/sql_show.cc

```diff
@@ -5310,6 +5310,8 @@ int get_all_tables(THD *thd, TABLE_LIST *tables, COND *cond)
       }
 
       TABLE *tmp_tbl= share_temp->all_tmp_tables.front();
+      if(!tmp_tbl)
+        continue;
```

Comment:
No, this is completely wrong. There cannot be a temporary table **share** without a single temporary **table**.

If this is the case here — it's a bug.